pkg/ingester/ingester.go

```diff
@@ -474,6 +476,13 @@ func (u *userTSDB) blocksToDelete(blocks []*tsdb.Block) map[ulid.ULID]struct{} {
 		return deletable
 	}
 
+	now := time.Now().UnixMilli()
+	for _, b := range blocks {
+		if now-b.MaxTime() >= u.blockRetentionPeriod {
```

Comment:
Why we don't apply the same if shipper is disabled

Reply:
Fixed